### PR TITLE
devsim: add desktop portability devsim config files

### PR DESCRIPTION
**Pull request by christophe-lunarg**
I created https://docs.google.com/spreadsheets/d/1QCfIMaq7dV7hZWrn0_NU7uVQvuph4BI0cpMInHfosns/edit#gid=0 to review the devsim support of Vulkan capbabilities.

A side effect is that it allowed me to create configuration files for desktop 1.0 (macOS, Linux and Windows) and desktop 1.2 (Linux and Windows only as MoltenVK doesn't support 1.2).

The minimum requirements remains perfectly relevant for this release as it allows the Vulkan developers to judge the gap between the effective requirements in a given ecossytem and the minimum requirements.

### CODE REVIEW

File: layersvt/VkLayer_device_simulation.json.in [hunk 1]
```diff
@@ -21,8 +21,48 @@
         "features": {
             "presets": [
                 {
-                    "label": "macOS portability",
-                    "description": "Check the application for Vulkan portability subset.",
+                    "label": "Desktop Vulkan 1.0 Portability",
```
Comment (jeremyk-lunarg) [inline layersvt/VkLayer_device_simulation.json.in:24]: The preset is labeled as "Portability", but the config file it uses is called "common_desktop_1_0.json".

Is this common portability minimums, or common desktop minimums?

File: layersvt/VkLayer_device_simulation.json.in [hunk 2]
```diff
@@ -21,8 +21,48 @@
...
+                            "value": "whitelist"
+                        },
+                        {
+                            "key": "filename",
+                            "value": "${VULKAN_CONTENT}/VK_LAYER_LUNARG_device_simulation/common_desktop_1_0.json"
+                        }
+                    ]
+                },
+                {
+                    "label": "Desktop Vulkan 1.2 Portability",
```
Comment (jeremyk-lunarg) [inline layersvt/VkLayer_device_simulation.json.in:44]: Same question as above, but with 1.2 instead of 1.0.

**Review by jeremyk-lunarg — CHANGES_REQUESTED** [review]
A few semantic questions, and there are some merge conflicts that need fixing.

Also, is there a reason this PR excludes a 1.1 config?

File: layersvt/VkLayer_device_simulation.json.in (hunk 1 above)
Comment (christophe-lunarg) [reply layersvt/VkLayer_device_simulation.json.in:24]: True I should make it more consistent. 

It's not the minimums, it's effectively what's available in the ecosystem.

**Comment by christophe-lunarg** [comment]
I didn't excluded Vulkan 1.1, I could not find significant macOS data to make a relevant Vulkan 1.1 file.

File: layersvt/VkLayer_device_simulation.json.in (hunk 1 above)
Comment (jeremyk-lunarg) [reply layersvt/VkLayer_device_simulation.json.in:24]: So if I'm understanding the name changes correctly, these are config files with features and limits common between fully conformant desktop implementations, and NOT portability subsets?

Comment (christophe-lunarg) [reply layersvt/VkLayer_device_simulation.json.in:24]: Yes it's "fully conformant desktop implementations" but that include portability subset as it's required for cross platform support (macOS).

**Review by jeremyk-lunarg — APPROVED** [review]
LGTM

**Pull request merged**